### DISCUSSION TITLE
did:comos method draft, and how can we make it compatible with HID-Network?

### DISCUSSION THREAD

**Discussion by Vishwas1** [hypersign-protocol/hid-node#127] [Ideas]
- https://github.com/hypersign-protocol/did-cosmos/blob/main/README.md 
- https://github.com/cosmos/chain-registry 
- https://github.com/hypersign-protocol/did-cosmos/blob/main/README.md#adding-support-for-didcosmos

**Comment by arnabghose997** [comment]
- IID are basically DID with additional properties.
- The IID seems to be restricted to on-chain assets such as fungible tokens, non-fungible tokens  and other chain-native components.
    - They dont refer real world objects like poeple
    - The on-chain assets, that the IIDs point to, can have reference to real-world objects. For Example- an NFT’s creator
- There are three modules (or segments) that are required:
    - **Chain Registry**: It has a set of chain descriptors which tells us how access different chains, using their **chainspace**
    - **Namespace Registry:** It describes how to access particular module assets, using their **namespace**
    - **Asset Module:** Maintains on-chain assets, and support functions such as Create, Update, Deactivate and Retrieve
- There are four forms of did:cosmos DIDs:

```jsx
did:cosmos:version:chainspace:namespace:unique-id
did:cosmos:chainspace:namespace:unique-id
did:cosmos:version:chainspace
did:cosmos:chainspace
```

- Components of DID:
    - `version` - It should be an integer. With every breaking change, it should increment. This will also ensure backward compatibility. If the `version` component isn’t present, it must be assumed version 1.
    - `chainspace` - It is used to identify the blockchain (”cosmos”, “ixo”, etc)
    - `asset-id` - Unique identifier for on-chain assets. It comprises of `namespace` and `unique-id`
        - `namespace` - It points various namespaces handled by modules such as “nft”, “bank”, “staking”. It routes incoming resolution request to correct asset module.
        - `unique-id` - It uniquely identifies an asset. It MUST be a representation of Secp256k1 public key ([https://w3c-ccg.github.io/did-method-key/#secp256k1](https://w3c-ccg.github.io/did-method-key/#secp256k1))
- If `asset-id` is not present in DID, it refers to the chainDescriptor which specifies the chain from chain-registry. For example, `did:cosmos:ixo` refers to the chain descriptor associated with the "ixo" chain space in the chain registry.****
- Example: `did:cosmos:1:ixo:nft:7Tqg6BwSSWapxgUDm9KKgg`

`did:cosmos` - DID Method
`1` - Version
`ixo` - chainspace
`nft` - namespace
`7Tqg6BwSSWapxgUDm9KKgg` - asset-id
- Adding support for `did:cosmos`
    - The chain must be registered with Cosmos Registry (https://github.com/cosmos/chain-registry)
- IIDs have additional properties apart from the usual DID Core Spec properties:
    - **LinkedResource**
        - its a privacy-enabled way to attach resources to on-chain assets
        - Example
        
        ```jsx
        {
         "linkedResource": [{
        
          "path"(optional): // IID Resource path for this resource in the asset namespace, e.g., /myResource.png
        
           "id"(optional): // IID Reference ID for this resource, e.g., #myResource.png
        
           "rel"(optional): // the relationship of this resource to the IID Asset
        
           "type": "nft:ResourceDescriptor", // The JSON-LD type of this resource
        
          "proof": [ // an array of proofs, any of which may be used
           {
            "type"(optional): "hash" | "hashgraph" | "hashset" // the form of proof used to verify the resource as authentic
            "stage"(optional): "raw" | "compressed" | "encrypted" | "encoded" // the 
            "value"(optional): hash | hashgraph | hashset, // the actual proof for this particular resource
           }
          ],
          "resourceFormat": media type, // the IANA media-type of the linked resource
        
          "compression"(optional): "gzip" | "none", // the compression (performed on the raw resource) of an inline resource
        
          "encryption": [open ended
           for arbitrary extensibility
          ], // the encryption technique applied after compression and before encoding
        
          "encoding"(optional): "native" | "multibase" | "string", // the encoding (after compression and encryption) of an inline resource; "native" means the inline resource is native JSON-LD with neither compression nor encryption
        
          "endpoint"(optional): url, // a URL at which this resource can be retrieved before decrypting and decrompressing
        
          "resource"(optional): "..."
          a representation of the resource
          for inline
          distribution,
          encoded as per "encoding"
            }]
        }
        ```
- As Hypersign Identity Network's DID are used to represent people who create their decentralised identity, which seems different from what IIDs are representative of.
